### PR TITLE
Add webpack directory aliasing (config changes)

### LINKED ISSUE #2567: Add webpack directory aliasing
**Is your feature request related to a problem? Please describe.**
As simorgh gets bigger we are ending up with imports like `'../../../../lib/analyticsUtils'` which isnt very readable at all.

**Describe the solution you'd like**
Add webpack aliases so rather than `'../../../../lib/analyticsUtils'` you can do `'@lib/analyticsUtils'`

Possible other ones include `@components` `@containers` `@contexts` `@testHelpers` `@app` `@server` 

**Testing notes**
[Tester to complete]

Dev insight: Just a sanity check that it builds fine, pages load all correct stuff etc

**Additional context**

### PR DESCRIPTION
**Pull request by oluoluoxenfree**
Part completes #2567 

**Overall change:** _A very high-level summary of easily-reproducible changes that can be understood by non-devs._

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -33,7 +33,12 @@ module.exports = (shell = {}) => {
   const baseConfig = {
     mode: IS_PROD ? 'production' : 'development',
     devtool: IS_PROD ? 'source-map' : 'cheap-eval-source-map',
-    resolve: { extensions: ['.js', '.jsx'] }, // resolves `import '../Foo'` to `../Foo/index.jsx`
+    resolve: {
+      extensions: ['.js', '.jsx'],
+      alias: {
+        lib: resolvePath('src/app/lib/'),
```
Comment (dr3) [inline webpack.config.js]:
```suggestion
        '@lib': resolvePath('src/app/lib/'),
```

**Review by dr3 — COMMENTED** [review]
Hey, this looks great, can I be a pain and ask we switch from `@` to `#`'s though :P (soz i know i suggested `@`'s originally :P ) Think now they will reduce confusions with npm packages.

After that, LGTM :)

File: webpack.config.js [hunk 2]
```diff
@@ -33,7 +33,17 @@ module.exports = (shell = {}) => {
...
+      extensions: ['.js', '.jsx'],
+      alias: {
+        '@components': resolvePath('src/app/components'),
+        '@data': resolvePath('data/'),
+        '@lib': resolvePath('src/app/lib/'),
+        '@testHelpers': resolvePath('src/testHelpers/'),
+        '@app': resolvePath('src/app'),
+        '@contexts': resolvePath('src/app/contexts'),
+      },
+    }, // resolves `import '../Foo'` to `../Foo/index.jsx`
```
Comment (dr3) [inline webpack.config.js]: Can you pop this comment on line 37 please :)

File: package.json [hunk 3]
```diff
@@ -82,6 +82,7 @@
     "core-js": "^3.2.1",
     "dotenv": "^8.1.0",
     "es6-promise": "^4.2.8",
+    "eslint-import-resolver-webpack": "^0.11.1",
```
Comment (dr3) [inline package.json]: Not sure what this dep is doing, seems to work fine without it? Regardless needs to be a dev dep :)

**Review by dr3 — APPROVED** [review]

**Comment by dr3** [comment]
Have changed `Resolves #2567`  to `Part completes #2567` so this PR doesnt close this issue when merged, as a follow up PR is needed to use the new aliases :)

File: webpack.config.js [hunk 4]
```diff
@@ -33,7 +33,17 @@ module.exports = (shell = {}) => {
   const baseConfig = {
     mode: IS_PROD ? 'production' : 'development',
     devtool: IS_PROD ? 'source-map' : 'cheap-eval-source-map',
-    resolve: { extensions: ['.js', '.jsx'] }, // resolves `import '../Foo'` to `../Foo/index.jsx`
+    resolve: {
+      extensions: ['.js', '.jsx'], // resolves `import '../Foo'` to `../Foo/index.jsx`
+      alias: {
+        '#components': resolvePath('src/app/components'),
```
Comment (jamesbhobbs) [inline webpack.config.js]: I'm confused by the choices, why not containers?
Also can we alphabetise pls

**Review by jamesbhobbs — DISMISSED** [review]

File: webpack.config.js (hunk 4 above)
Comment (jamesbhobbs) [reply webpack.config.js]: Can you clarify the reasoning for inclusion?

**Review by Bopchy — COMMENTED** [review]
Looks good - though I'm also curious about why some things were aliased and not others. I'm assuming that the focus is on the modules that have lengthy paths & most accessed, and if so - I think `src/app/models` should be aliased as well

**Review by dr3 — APPROVED** [review]

**Review by Bopchy — APPROVED** [review]

**Pull request merged**